cloud/services/domains.go

```diff
@@ -85,46 +82,43 @@ func DeleteIPFromDNS(ctx context.Context, mscope *scope.MachineScope) error {
 	domainHostname := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier
 
 	// Delete A record
-	for ipType, publicIP := range publicIPs {
-		if err := DeleteDomainRecord(ctx, mscope, domainHostname, publicIP, dnsTTLSec, domainID, ipTypeToRecordTypeMapper[ipType]); err != nil {
-			return err
+	for _, publicIPs := range publicIPMapper {
+		for ipType, publicIP := range publicIPs {
+			if err := DeleteDomainRecord(ctx, mscope, domainHostname, publicIP, dnsTTLSec, domainID, ipTypeToRecordTypeMapper[ipType]); err != nil {
+				return err
+			}
+			// Delete TXT record
+			if err := DeleteDomainRecord(ctx, mscope, domainHostname, publicIP, dnsTTLSec, domainID, "TXT"); err != nil {
+				return err
+			}
 		}
 	}
 
-	// Delete TXT record
-	txtRecordValueString := createSHA256HashOfString(mscope.LinodeMachine.Name)
-	if err := DeleteDomainRecord(ctx, mscope, domainHostname, "owner:"+txtRecordValueString, dnsTTLSec, domainID, "TXT"); err != nil {
-		return err
-	}
-
 	// Wait for TTL to expire
 	time.Sleep(time.Duration(dnsTTLSec) * time.Second)
 
 	return nil
 }
 
 // getMachinePublicIPs gets the machines public IP
-func getMachinePublicIPs(ctx context.Context, mscope *scope.MachineScope) (map[string]string, error) {
-	// Verify instance id is not nil
-	if mscope.LinodeMachine.Spec.InstanceID == nil {
-		return nil, errors.New("instance ID is nil. cant get machine's public ip")
-	}
-
-	// Get the public IP that was assigned
-	addresses, err := mscope.LinodeClient.GetInstanceIPAddresses(ctx, *mscope.LinodeMachine.Spec.InstanceID)
-	if err != nil {
-		return nil, err
-	}
-
-	if len(addresses.IPv4.Public) == 0 || addresses.IPv6 == nil {
-		return nil, errors.New("no public address")
-	}
-
-	if addresses.IPv6.SLAAC == nil {
-		return nil, errors.New("no SLAAC address")
+func getMachinePublicIPs(mscope *scope.MachineScope) ([]map[string]string, error) {
+	if mscope.LinodeMachine.Status.Addresses == nil {
+		return nil, fmt.Errorf("no addresses available on the LinodeMachine resource")
+	}
+	IPsToReturn := make([]map[string]string, 0, 2) //nolint:mnd // Max 2 IPs(1 IPv4 and 1 IPv6)
+	for _, IPs := range mscope.LinodeMachine.Status.Addresses {
+		publicIPs := make(map[string]string)
+		if IPs.Type == "ExternalIP" {
+			if net.ParseIP(IPs.Address).To4() != nil {
```

Comment:
If you can, please use the `net/netip` package for handling IP addresses. It has a much less-error-prone API.

Reply:
Aye! Updated the code to use `net/netip`